zerver/lib/message.py

```diff
@@ -189,18 +191,31 @@ def wide_dict(message: Message) -> Dict[str, Any]:
         return obj
 
     @staticmethod
-    def post_process_dicts(objs: List[Dict[str, Any]], apply_markdown: bool, client_gravatar: bool) -> None:
+    def post_process_dicts(objs: List[Dict[str, Any]], apply_markdown: bool, client_gravatar: bool) -> \
+            List[Dict[str, Any]]:
```

Comment:
I think with `post_process_dicts` our intention is to mutate these in place, for performance reasons.  That function is only used in the message-fetch codepath, where each message is only touched once.  That was the reason I suggested that we have `_finalize_payload_in_place`.

Reply:
So, instead of calling `finalize_payload` we can call `_finalize_payload` instead?